pallets/usdv-minting/src/lib.rs

```diff
@@ -42,14 +47,34 @@ decl_module! {
         fn deposit_event() = default;
 
         #[weight = 0]
-        pub fn mint(origin, balance: BalanceOf<T>) -> dispatch::DispatchResult{
-            let provider = ensure_signed(origin)?;
+        pub fn mint(
+            origin,
+            asset_id: Asset,
+            balance: BalanceOf<T>,
+            provider_member: <T as frame_system::Trait>::AccountId
+        ) -> dispatch::DispatchResult {
+            let mint_member = ensure_signed(origin)?;
+
+            pallet_membership::Module::<T, MintMembers>::members()
+                .binary_search(&mint_member)
+                .ok()
+                .ok_or(pallet_membership::Error::<T, MintMembers>::NotMember)?;
+
+            pallet_membership::Module::<T, ProviderMembers>::members()
+                .binary_search(&provider_member)
+                .ok()
+                .ok_or(pallet_membership::Error::<T, ProviderMembers>::NotMember)?;
+
+            let curr_asset_amount = T::Currency::reserved_balance(asset_id, &provider_member);
+            let curr_usdv_amount = T::Currency::free_balance(Asset::Usdv, &provider_member);
+            let diff_amount = curr_asset_amount - curr_usdv_amount;
 
-            let members = pallet_membership::Module::<T, Instance0>::members();
-            members.binary_search(&provider).ok().ok_or(pallet_membership::Error::<T, Instance0>::NotMember)?;
+            if balance > diff_amount {
+                return Err(Error::<T>::InsufficientFunds.into());
+            }
 
-            T::Currency::deposit(Asset::Usdv, &provider, balance)?;
-            Self::deposit_event(RawEvent::Mint(provider, Asset::Usdv));
+            T::Currency::deposit(Asset::Usdv, &provider_member, balance)?;
+            Self::deposit_event(RawEvent::Mint(mint_member, Asset::Usdv));
```

Comment:
Basically searches if a given provider has funds that can be converted into USDv. Since "attested" assets are already locked by default, this function simply checks for funds and adds Valiu Dolars accordingly